src/js/common/schemaform/helpers.js

```diff
@@ -303,7 +303,15 @@ export function getNonArraySchema(schema) {
     }
 
     if (newProperties !== schema.properties) {
-      return _.set('properties', newProperties, schema);
+      let newSchema = _.set('properties', newProperties, schema);
+      if (newSchema.required) {
+        const newRequired = _.intersection(Object.keys(newSchema.properties), newSchema.required);
+        if (newRequired.length !== newSchema.required.length) {
+          newSchema = _.set('required', newRequired, newSchema);
```

Comment:
This is due to how we deal with array fields on the review page. We pull them out of the page they're in and display them separately. Before, we were doing that but not removing the array fields from the required array, which caused the phantom validation error you saw. I won't cause any submission errors because we still validate everything before sending when you click submit. This change doesn't get saved anywhere, it's a temporary schema that's only used in the components on the review page.

It is another edge case, though, when you have an array field that requires at least one item. It's not really clear that the user has to enter an item if they come to the review page without one already. I opened an issue for us to think about simplifying how we handle array fields on the review page.